providers/openshift/provider.go

```diff
@@ -131,18 +138,29 @@ func (p *OpenShiftProvider) newOpenShiftClient() (*http.Client, error) {
 		capaths = paths
 		system_roots = false
 	}
+
+	// try to retrieve a cached client
```

Comment:
A unit test should be able to prove if this works or not.  i.e. run it twice assert you get same client.  The change the input paths and assert you get a different client.

Reply:
Good idea :+1: